tools/testsys/src/aws_ecs.rs

```diff
@@ -0,0 +1,173 @@
+use crate::aws_resources::{ami, ec2_crd, get_ami_id, migration_crd};
+use crate::crds::{
+    BottlerocketInput, CrdCreator, CrdInput, CreateCrdOutput, MigrationInput, OrchestratorInput,
+    TestInput,
+};
+use crate::error::{self, Result};
+use bottlerocket_types::agent_config::{ClusterType, EcsClusterConfig, EcsTestConfig};
+use log::debug;
+use maplit::btreemap;
+use model::{Crd, DestructionPolicy};
+use snafu::OptionExt;
+
+/// A `CrdCreator` responsible for creating crd related to `aws-ecs` variants.
+pub(crate) struct AwsEcsCreator {
+    pub(crate) region: String,
+    pub(crate) ami_input: String,
+    pub(crate) migrate_starting_commit: Option<String>,
+}
+
+#[async_trait::async_trait]
+impl CrdCreator for AwsEcsCreator {
+    /// Determine the AMI from `amis.json`.
+    fn image_id(&self, _: &CrdInput) -> Result<String> {
+        ami(&self.ami_input, &self.region)
+    }
+
+    /// Determine the starting image from EC2 using standard Bottlerocket naming conventions.
+    async fn starting_image_id(&self, crd_input: &CrdInput) -> Result<String> {
+        get_ami_id(
+            format!(
+                "bottlerocket-{}-{}-{}-{}",
+                crd_input.variant, crd_input.arch, crd_input.starting_version.as_ref().context(error::InvalidSnafu{what: "The starting version must be provided for migration testing"})?, self.migrate_starting_commit.as_ref().context(error::InvalidSnafu{what: "The commit for the starting version must be provided if the starting image id is not"})?
```

Comment:
Can you wrap this? Sadly rustfmt can't seem to handle things inside of macro calls.